From 6dd8fde991692981126fefd09198300a8aaa9849 Mon Sep 17 00:00:00 2001
From: Faouaz TENOUTIT <faouaz.tenoutit@intel.com>
Date: Tue, 18 Nov 2014 11:02:13 -0800
Subject: [PATCH 1/1] ASoC: Intel: Fix wrong condition test

snd_soc_sst_platform: sst_handle_vb_timer: failed to set recovery timer

This is due to wrong condition test in the driver (should be < instead of != 0).

Issue: IRDA-1685
Change-Id: Ie8704c223ccd8246c130995084e2b8e66a648455
Signed-off-by: Eric Ernst <eric.ernst@intel.com>
Signed-off-by: Faouaz TENOUTIT <faouaz.tenoutit@intel.com>
---
 sound/soc/intel/platform-libs/controls_v2_dpcm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 12aaf9f06368..fd202e16191c 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -859,7 +859,7 @@ void sst_handle_vb_timer(struct snd_soc_platform *p, bool enable)
 				SST_TASK_SBA, 0, &cmd, sizeof(cmd.header) + cmd.header.length) == 0) {
 
 			if (sst_dsp->ops->set_generic_params(SST_SET_MONITOR_LPE,
-									(void *)&enable) != 0)
+									(void *)&enable) < 0)
 				pr_err("%s: failed to set recovery timer\n", __func__);
 		} else
 			pr_err("%s: failed to send sst cmd %d\n",
-- 
1.7.9.5

